include/z64collision_check.h

```diff
@@ -9,245 +9,382 @@
+} ColliderShape;
+
+/**
+ * Affects the sound Link's sword makes when hitting it, hookability,
+ * and possibly other things. It's definitely not flags, as all checks
+ * are == or !=. Will probably need more actors decomped to truly 
+ * understand what this is.
+ */
+typedef enum ElementType {
+    /* 0 */ ELEMTYPE_UNK0,
+    /* 1 */ ELEMTYPE_UNK1,
+    /* 2 */ ELEMTYPE_UNK2,
+    /* 3 */ ELEMTYPE_UNK3,
+    /* 4 */ ELEMTYPE_UNK4,
+    /* 5 */ ELEMTYPE_UNK5,
+    /* 6 */ ELEMTYPE_UNK6,
+    /* 7 */ ELEMTYPE_UNK7
+} ElementType;
+
+#define AT_NONE 0 // No flags set. Cannot have AT collisions when set as AT
+#define AT_ON (1 << 0) // Can have AT collisions when set as AT
+#define AT_HIT (1 << 1) // Had an AT collision
+#define AT_BOUNCED (1 << 2) // Had an AT collision with an AC_HARD collider
+#define AT_TYPE_PLAYER (1 << 3) // Has player-aligned damage
+#define AT_TYPE_ENEMY (1 << 4) // Has enemy-aligned damage
+#define AT_TYPE_OTHER (1 << 5) // Has non-aligned damage
+#define AT_SELF (1 << 6) // Can have AT collisions with colliders attached to the same actor
+#define AT_TYPE_ALL (AT_TYPE_PLAYER | AT_TYPE_ENEMY | AT_TYPE_OTHER) // Has all three damage alignments
+
+#define AC_NONE 0 // No flags set. Cannot have AC collisions when set as AC
+#define AC_ON (1 << 0) // Can have AC collisions when set as AC
+#define AC_HIT (1 << 1) // Had an AC collision
+#define AC_HARD (1 << 2) // Causes AT colliders to bounce off it
+#define AC_TYPE_PLAYER AT_TYPE_PLAYER // Takes player-aligned damage
+#define AC_TYPE_ENEMY AT_TYPE_ENEMY // Takes enemy-aligned damage
+#define AC_TYPE_OTHER AT_TYPE_OTHER // Takes non-aligned damage
+#define AC_NO_DAMAGE (1 << 6) // Collider does not take damage
+#define AC_BOUNCED (1 << 7) // Caused an AT collider to bounce off it
+#define AC_TYPE_ALL (AC_TYPE_PLAYER | AC_TYPE_ENEMY | AC_TYPE_OTHER) // Takes damage from all three alignments
+
+#define OC1_NONE 0 // No flags set. Cannot have OC collisions when set as OC
+#define OC1_ON (1 << 0) // Can have OC collisions when set as OC
+#define OC1_HIT (1 << 1) // Had an OC collision
+#define OC1_NO_PUSH (1 << 2) // Does not push other colliders away during OC collisions
+#define OC1_TYPE_PLAYER (1 << 3) // Can have OC collisions with OC type player
+#define OC1_TYPE_1 (1 << 4) // Can have OC collisions with OC type 1
+#define OC1_TYPE_2 (1 << 5) // Can have OC collisions with OC type 2
+#define OC1_TYPE_ALL (OC1_TYPE_PLAYER | OC1_TYPE_1 | OC1_TYPE_2) // Can have collisions with all three OC types
+
+#define OC2_NONE 0 // No flags set. Has no OC type
+#define OC2_HIT_PLAYER (1 << 0) // Had an OC collision with OC type player
+#define OC2_UNK1 (1 << 1) // Prevents OC collisions with OC2_UNK2. Some horses and toki_sword have it.
+#define OC2_UNK2 (1 << 2) // Prevents OC collisions with OC2_UNK1. Nothing has it.
+#define OC2_TYPE_PLAYER (1 << 3) // Has OC type player
+#define OC2_TYPE_1 (1 << 4) // Has OC type 1
+#define OC2_TYPE_2 (1 << 5) // Has OC type 2
```

Comment:
I think we also wanted to do this:
```suggestion
#define OC2_TYPE_PLAYER OC1_TYPE_PLAYER // Has OC type player
#define OC2_TYPE_1 OC1_TYPE_1 // Has OC type 1
#define OC2_TYPE_2 OC1_TYPE_2 // Has OC type 2
```